lib/std/io_uring.zig

```diff
@@ -0,0 +1,841 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2015-2020 Zig Contributors
+// This file is part of [zig](https://ziglang.org/), which is MIT licensed.
+// The MIT license requires this copyright notice to be included in all copies
+// and substantial portions of the software.
+const builtin = @import("builtin");
+const std = @import("std");
+const assert = std.debug.assert;
+const os = std.os;
+const linux = os.linux;
+const mem = std.mem;
+const net = std.net;
+const testing = std.testing;
+
+pub const io_uring_params = linux.io_uring_params;
+pub const io_uring_cqe = linux.io_uring_cqe;
+
+// TODO Update linux.zig's definition of linux.io_uring_sqe:
+// linux.io_uring_sqe uses numbered unions, i.e. `union1` etc. that are not future-proof and need to
+// be re-numbered whenever new unions are interposed by the kernel. Furthermore, Zig's unions do not
+// support assignment by any union member directly as in C, without going through the union, so the
+// kernel adding new unions would also break existing Zig code.
+// We therefore use a flat struct without unions to avoid these two issues.
+// Pending https://github.com/ziglang/zig/issues/6349.
+pub const io_uring_sqe = extern struct {
+    opcode: linux.IORING_OP,
+    flags: u8 = 0,
+    ioprio: u16 = 0,
+    fd: i32 = 0,
+    off: u64 = 0,
+    addr: u64 = 0,
+    len: u32 = 0,
+    opflags: u32 = 0,
+    user_data: u64 = 0,
+    buffer: u16 = 0,
+    personality: u16 = 0,
+    splice_fd_in: i32 = 0,
+    options: [2]u64 = [2]u64{ 0, 0 }
+};
+
+// TODO Add to zig/std/os/bits/linux.zig:
+const IORING_SQ_CQ_OVERFLOW = 1 << 1;
+
+comptime {
+    assert(@sizeOf(io_uring_params) == 120);
+    assert(@sizeOf(io_uring_sqe) == 64);
+    assert(@sizeOf(io_uring_cqe) == 16);
+
+    assert(linux.IORING_OFF_SQ_RING == 0);
+    assert(linux.IORING_OFF_CQ_RING == 0x8000000);
+    assert(linux.IORING_OFF_SQES == 0x10000000);
+}
+
+pub const IO_Uring = struct {
```

Comment:
I think for now the style guide is `IoUring`

Reply:
Anticipates https://github.com/ziglang/zig/issues/1097?

Reply:
It is not yet accepted